v2/api/apimanagement/v1api20220801/structure.txt

```diff
@@ -2,6 +2,133 @@
 github.com/Azure/azure-service-operator/v2/api/apimanagement/v1api20220801
 ├── APIVersion: Enum (1 value)
 │   └── "2022-08-01"
+├── Backend: Resource
+│   ├── Owner: Service
+│   ├── Spec: Object (11 properties)
+│   │   ├── AzureName: Validated<string> (2 rules)
+│   │   │   ├── Rule 0: MaxLength: 80
+│   │   │   └── Rule 1: MinLength: 1
+│   │   ├── Credentials: *Object (5 properties)
+│   │   │   ├── Authorization: *Object (2 properties)
+│   │   │   │   ├── Parameter: Validated<*string> (2 rules)
+│   │   │   │   │   ├── Rule 0: MaxLength: 300
+│   │   │   │   │   └── Rule 1: MinLength: 1
+│   │   │   │   └── Scheme: Validated<*string> (2 rules)
+│   │   │   │       ├── Rule 0: MaxLength: 100
+│   │   │   │       └── Rule 1: MinLength: 1
+│   │   │   ├── Certificate: Validated<string[]> (1 rule)
```

Comment:
Confirmed with @matthchr that the `Certificate` property requires a Client Certificate Thumbprint. So it should be ok to ignore this one.

Reply:
I agree that it's not a secret - but should probably be marked as configurable so that a new cert can be rotated into place via a conflig change, instead of requiring an update of the entire resource.

Reply:
The `Certificate` property is an array. If I attempt the following

```yaml
      BackendCredentialsContract:
        Certificate:
          $importConfigMapMode: optional
```
I get this error
```
error generating code: failed to execute stage 35: Replace properties flagged as a configMap with genruntime.ConfigMapReference. For properties flagged as an optional configMap, add a new <property>FromConfig property.: transforming spec secrets: visiting type \"github.com/Azure/azure-service-operator/v2/api/apimanagement/v1api20220801/BackendCredentialsContract\": visit of type of \"github.com/Azure/azure-service-operator/v2/api/apimanagement/v1api20220801/BackendCredentialsContract\" failed: failed to transform property to optional configmap on type github.com/Azure/azure-service-operator/v2/api/apimanagement/v1api20220801/BackendCredentialsContract: expected property \"Certificate\" to be a string, but was: Validated[Array[string]]"
```

Reply:
Resolving because this is a Thumbprint